internal/cmd/agent/register/register.go

```diff
@@ -59,6 +59,24 @@ func Get(ctx context.Context, namespace string, cfg *rest.Config) (*AgentInfo, e
 		return nil, err
 	}
 
+	configMap, err := k8s.Core().V1().ConfigMap().Get(namespace, config.AgentConfigName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		logrus.Warnf("Cannot find %s config map", config.AgentConfigName)
+
+		return nil, err
+	}
+
+	agentConfig, err := config.ReadConfig(configMap)
+	if err != nil {
+		logrus.Warnf("Cannot parse %s config map into agent config", config.AgentConfigName)
+
+		return nil, err
+	}
```

Comment:
`runRegistration` loads it like this:

```suggestion
        cfg, err := config.Lookup(ctx, namespace, config.AgentConfigName, k8s.Core().V1().ConfigMap())
	if err != nil {
		return nil, fmt.Errorf("failed to look up client config %s/%s: %w", namespace, config.AgentConfigName, err)
	}

```

Reply:
That's much better :) Thanks, done.